src/functional/ffront/func_to_foast.py

```diff
@@ -327,28 +395,73 @@ def visit_Lt(self, node: ast.Lt, **kwargs) -> foast.CompareOperator:
     def visit_Eq(self, node: ast.Eq, **kwargs) -> foast.CompareOperator:
         return foast.CompareOperator.EQ
 
+    def _verify_builtin_function(self, node: ast.Call):
+        func_name = self._func_name(node)
+        func_info = getattr(fbuiltins, func_name).__gt_type__()
+        if not len(node.args) == len(func_info.args):
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{func_name}() expected {len(func_info.args)} positional arguments, {len(node.args)} given!",
+            )
+        elif unexpected_kwargs := set(k.arg for k in node.keywords) - set(func_info.kwargs):
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() got unexpected keyword arguments: {unexpected_kwargs}!",
+            )
+
+    def _verify_builtin_type_constructor(self, node: ast.Call):
+        if not len(node.args) == 1:
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() expected 1 positional argument, {len(node.args)} given!",
+            )
+        elif node.keywords:
+            unexpected_kwargs = set(k.arg for k in node.keywords)
+            raise FieldOperatorSyntaxError.from_AST(
+                node,
+                msg=f"{self._func_name(node)}() got unexpected keyword arguments: {unexpected_kwargs}!",
+            )
```

Comment:
Will do.

Reply:
Actually It doesn't look like it. The mentioned method takes a list of `SymbolTypes` for args and all I have is python ast nodes. I could visit them but at this point not all of them have types attached. The underlying function considers incomplete types as incompatible.

I vote to leave it like this for now because to make that method reusable it would have to be essentially rewritten.

Reply:
I agree, seperate project then.